test/api3-server-v1/proxies/ProxyFactoryZkSync.sol.js

```diff
@@ -0,0 +1,239 @@
+    const api3ServerV1Factory = await ethers.getContractFactory('Api3ServerV1', roles.deployer);
+    const api3ServerV1 = await api3ServerV1Factory.deploy(
+      accessControlRegistry.address,
+      api3ServerV1AdminRoleDescription,
+      roles.manager.address
+    );
+    const proxyFactoryFactory = await ethers.getContractFactory('ProxyFactoryZkSync', roles.deployer);
+    const proxyFactory = await proxyFactoryFactory.deploy(api3ServerV1.address);
+
+    const endpointId = testUtils.generateRandomBytes32();
+    const templateParameters = testUtils.generateRandomBytes();
+    const templateId = ethers.utils.keccak256(
+      ethers.utils.solidityPack(['bytes32', 'bytes'], [endpointId, templateParameters])
+    );
+    const beaconId = ethers.utils.keccak256(
+      ethers.utils.solidityPack(['address', 'bytes32'], [roles.airnode.address, templateId])
+    );
+    await api3ServerV1.connect(roles.manager).setDapiName(dapiName, beaconId);
+
+    const beaconValue = 123;
+    const beaconTimestamp = await helpers.time.latest();
+    const data = ethers.utils.defaultAbiCoder.encode(['int256'], [beaconValue]);
+    const signature = await testUtils.signData(roles.airnode, templateId, beaconTimestamp, data);
+    await api3ServerV1.updateBeaconWithSignedData(roles.airnode.address, templateId, beaconTimestamp, data, signature);
+
+    return {
+      roles,
+      api3ServerV1,
+      proxyFactory,
+      dapiName,
+      dapiNameHash,
+      beaconId,
+      beaconValue,
+      beaconTimestamp,
+    };
+  }
+
+  describe('Proxy contract hashes', function () {
+    it('are correct', async function () {
+      const fs = require('fs');
+      Object.keys(zkSyncUtils.bytecodeHashes).map((contractName) => {
+        const bytecode = JSON.parse(
+          fs.readFileSync(
+            `./artifacts-zk/contracts/api3-server-v1/proxies/${contractName}.sol/${contractName}.json`,
+            'utf8'
+          )
+        ).bytecode;
+        const bytecodeHash = ethers.utils.hexlify(zkSync.utils.hashBytecode(bytecode));
+        expect(bytecodeHash).to.equal(zkSyncUtils.bytecodeHashes[contractName]);
+      });
+    });
+  });
+
+  describe('constructor', function () {
+    context('Api3ServerV1 addres is not zero', function () {
+      it('constructs', async function () {
+        const { api3ServerV1, proxyFactory } = await helpers.loadFixture(deploy);
+        expect(await proxyFactory.api3ServerV1()).to.equal(api3ServerV1.address);
+      });
+    });
+    context('Api3ServerV1 addres is zero', function () {
+      it('reverts', async function () {
+        const { roles } = await helpers.loadFixture(deploy);
+        const proxyFactoryFactory = await ethers.getContractFactory('ProxyFactory', roles.deployer);
+        await expect(proxyFactoryFactory.deploy(ethers.constants.AddressZero)).to.be.revertedWith(
+          'Api3ServerV1 address zero'
+        );
+      });
+    });
+  });
+
```

Comment:
Note that I skipped the deployment tests because on a regular hardhat network deployments are made to the regular CREATE2 addresses and not the ones that the compute functions expect. I could have set up a zksync local test environment but I didn't attempt it because I don't expect that to be a smooth process. Imo seeing one correct proxy address derivation on a public zkSync chain will be evidence enough that everything will work correctly.